optuna/distributions.py

```diff
@@ -333,22 +335,23 @@ def __init__(self, low: int, high: int, step: int = 1) -> None:
                 "(low={}, high={}).".format(low, high)
             )
 
-        if step <= 0:
-            raise ValueError(
-                "The `step` value must be non-zero positive value, but step={}.".format(step)
-            )
-
         if low < 1.0:
             raise ValueError(
                 "The `low` value must be equal to or greater than 1 for a log distribution "
                 "(low={}, high={}).".format(low, high)
             )
 
-        high = _adjust_int_uniform_high(low, high, step)
+        if step != 1:
+            warnings.warn(
+                "`step` accepts only `1`, so `step` is replaced with `1`. "
+                "`step` argument is deprecated and will be removed in the future. "
+                "The removal of this feature is currently scheduled for v4.0.0, "
+                "but this schedule is subject to change.",
+                DeprecationWarning,
```

Comment:
I'm sorry for asking you a change again, but could you use `FutureWarning` instead of `DeprecationWarning`? #1428 introduced it to notify the warning to users.
```suggestion
                FutureWarning,
```